Sources/Extensions/SwiftStdlib/ArrayExtensions.swift

```diff
@@ -231,6 +231,23 @@ public extension Array where Element: Equatable {
 		return indices
 	}
 
+    @discardableResult
+    /// SwifterSwift: Remove first instance of an element from array.
+    ///
+    ///        [1, 2, 2, 3, 4, 2, 5].removeElement(2) -> [1, 2, 3, 4, 2, 5]
+    ///        ["h", "e", "l", "l", "o"].removeElement("e") -> ["h", "l", "l", "o"]
+    ///
+    /// - Parameter element: element to remove.
+    /// - Returns: the removed element, otherwise nil.
+    public mutating func removeElement(_ element: Element) -> Element? {
+        if count > 0 {
```

Comment:
I would prefer to use a `guard` here

Reply:
You can make the method more concise:
```Swift
extension RangeReplaceableCollection where Element: Equatable {
    @discardableResult
    public mutating func removeFirst(element: Element) -> Element? {
        guard let index = index(where: { $0 == element }) else { return nil }
        return remove(at: index)
    }
}
```
Maybe making it an extension of `RangeReplaceableCollection` is a bit excessive... 😂

Reply:
@guykogus @jdisho @SD10  Actually, since the element is equatable we can use `index(of: Element)` instead of the where closure :))

Reply:
I'm still wondering why not just make it:
`removeFirst(where condition: (Element) -> Bool) -> Element?`
You can do anything you want with this

Reply:
```swift
let first = [1, 2, 3].removeFirst(where: {$0 == 2})
```

Reply:
@SD10 I agree, It can have a lot more of use cases :))  And also, if you still want the Equatable one, you can build it in terms of this just simply calling a closure that does equal-equal, as you mention

Reply:
Why not have both?

```Swift
extension Array {
    @discardableResult
    public mutating func removeFirst(where condition: (Element) -> Bool) -> Element? {
        guard let index = index(where: condition) else { return nil }
        return remove(at: index)
    }
}

extension Array where Element: Equatable {
    @discardableResult
    public mutating func removeFirst(element: Element) -> Element? {
        guard let index = index(of: element) else { return nil }
        return remove(at: index)
    }
}
```

Reply:
Seems like this PR has gone stale. I agree that `removeFirst(where:)` makes an existing task simpler, but TBH I don't see why we need both versions. It just adds boilerplate code to the library. 

Open to putting this up to a vote.
👍  for `removeFirst(where: (Element) -> Bool)`
👎  for  both `removeFirst(where: (Element) -> Bool)` && `removeFirst(element: Element)`

Reply:
cc @jdisho ☝️ If you're open to changing this to `removeFirst(where: throws (Element) -> Bool) -> rethrows Element?` we can get this one through

Reply:
Yes we can 😊 @SD10 
Btw, sorry for the delay, I will finish it asap.